packages/fiori/src/DynamicPage.ts

```diff
@@ -0,0 +1,338 @@
+ * </ul>
+ *
+ * <h3>Responsive Behavior</h3>
+ *
+ * Dynamic page web component implements the resposive paddings design.
+ *
+ * <h3>Keyboard Handling</h3>
+ *
+ * <h4>Basic Navigation</h4>
+ * <ul>
+ * <li>[SPACE, ENTER, RETURN] - If focus is on a button inside DynamicPageTitle its action is being triggered, once activated.
+ * If focus is on the snap header button (arrow button), or on the header itself, once activated, it triggers the associated action (such as snap/expand the header).
+ * If focus is on pin button (the button with pin icon on the bottom of the header), once activated, it triggers the associated action (pinning of the header). </li>
+ * </ul>
+ *
+ * <h4>Fast Navigation</h4>
+ * <ul>
+ * <li>This component provides a build in fast navigation group which can be used via <code>F6 / Shift + F6</code> or <code> Ctrl + Alt(Option) + Down /  Ctrl + Alt(Option) + Up</code>.
+ * In order to use this functionality, you need to import the following module:
+ * <code>import "@ui5/webcomponents-base/dist/features/F6Navigation.js"</code></li>
+ * </ul>
+ *
+ * <h3>ES6 Module Import</h3>
+ *
+ * <code>import "@ui5/webcomponents-fiori/dist/DynamicPage.js";</code>
+ *
+ * @constructor
+ * @extends UI5Element
+ * @since 1.23.0
+ * @public
+ */
+@customElement({
+	tag: "ui5-dynamic-page",
+	renderer: litRender,
+	styles: DynamicPageCss,
+	template: DynamicPageTemplate,
+	dependencies: [DynamicPageHeader, DynamicPageTitle, DynamicPageHeaderActions],
+})
+
+class DynamicPage extends UI5Element {
+	/**
+	 * Defines if the header is snapped.
+	 *
+	 * @default false
+	 * @public
+	 */
+	@property({ type: Boolean })
+	headerSnapped!: boolean;
+
+	/**
+	 * Defines if the header is pinned.
+	 *
+	 * @default false
+	 * @public
+	 */
+	@property({ type: Boolean })
+	headerPinned!: boolean;
+
+	/**
+	 * Defines if the footer is shown.
+	 *
+	 * @default false
+	 * @public
+	 */
+	@property({ type: Boolean })
+	showFooter!: boolean;
+
+	/**
+	 * Defines the current media query size.
+	 *
+	 * @private
+	 */
+	@property()
+	mediaRange!: string;
+
+	/**
+	 * Defines the content of the Dynamic Page.
+	 *
+	 * @public
+	 */
+	@slot({ "default": true, type: HTMLElement })
+	content!: HTMLElement[];
+
+	/**
+	 * Defines the title HTML Element.
+	 *
+	 * @public
+	 */
+	@slot({ type: DynamicPageTitle })
+	titleArea!: Array<DynamicPageTitle>;
+
+	/**
+	 * Defines the title HTML Element.
+	 *
+	 * @public
+	 */
+	@slot({ type: DynamicPageHeader })
+	headerArea!: Array<DynamicPageHeader>;
+
+	static i18nBundle: I18nBundle;
+
+	skipSnapOnScroll = false;
+	showHeaderInStickArea = false;
+	_updateMediaRange: ResizeObserverCallback;
+
+	constructor() {
+		super();
+
+		this._updateMediaRange = this.updateMediaRange.bind(this);
+	}
+
+	static async onDefine() {
+		DynamicPage.i18nBundle = await getI18nBundle("@ui5/webcomponents-fiori");
+	}
+
+	/**
+	 * Defines the title HTML Element.
+	 *
+	 * @public
+	 */
+	@slot({ type: HTMLElement })
+	footer!: HTMLElement[];
```

Comment:
Slot names could be re-discussed `titleArea`, `headerArea` and `footer`. Currently there is `titleArea`, but not `footerArea`. Let's think if we can go with just `title`, `header` and `footer`.

Reply:
The word "title" seems to be reserved for the HTMLElement's title attribute. We could rename the footer slot to footerArea, if those namings are also OK?